addons/sale_coupon/models/sale_coupon.py

```diff
@@ -115,8 +115,7 @@ def action_coupon_sent(self):
         }
 
     def action_coupon_cancel(self):
-        for coupon in self:
-            coupon.state = 'cancel'
+        self.state = 'cancel'
```

Comment:
This diff is not related to this task :p

Reply:
It is a different commit, and it still goes in the direction of the PR title ^^